src/lib/sw_engine/tvgSwRle.cpp

```diff
@@ -746,4 +834,58 @@ void rleFree(SwRleData* rle)
     if (!rle) return;
     if (rle->spans) free(rle->spans);
     free(rle);
-}
\ No newline at end of file
+}
+
+void updateRleSpans(SwRleData *rle, SwSpan* currentSpans, uint32_t newSize)
+{
+    SwSpan *spans = nullptr;
+    rle->size = newSize;
+    rle->spans = static_cast<SwSpan*>(realloc(rle->spans, rle->size * sizeof(SwSpan)));
+
+    spans = rle->spans;
+    for(int i = 0; i < (int)rle->size ; i++)
+    {
+       spans[i].x = currentSpans[i].x;
+       spans[i].y = currentSpans[i].y;
+       spans[i].len = currentSpans[i].len;
+       spans[i].coverage = currentSpans[i].coverage;
+    }
+}
+
+void rleClipPath(SwRleData *rle, const SwRleData *compRle)
+{
+    if (rle->size == 0 || compRle->size == 0) return;
+    uint32_t spanCount = rle->size > compRle->size ? rle->size : compRle->size ;
+    SwSpan *clipped = nullptr;
+    SwSpan *spans = rle->spans;
+    SwSpan *end = spans + rle->size;
+    SwSpan *cspans = nullptr;
+
+    cspans = (SwSpan*)malloc(sizeof(SwSpan) * (spanCount));
+    clipped = cspans;
+    _intersectSpansRegion(compRle, spans, end, &clipped, spanCount);
```

Comment:
auto clipped = _intersectSpansRect(compBBox, rle->spans, rle->spans + rle->size, spans, spanCnt);

Reply:
changed it